tests/Unit/Support/Traits/AssertCssTest.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace Pelago\Tests\Unit\Support\Traits;
+
+use Pelago\Tests\Unit\Support\Traits\Fixtures\TestingAssertCss;
+
+/**
+ * Test case.
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ */
+class AssertCssTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @test
+     */
+    public function getCssNeedleRegExpEscapesAllSpecialCharacters()
+    {
+        $needle = '.\\+*?[^]$(){}=!<>|:-/';
+
+        $result = TestingAssertCss::getCssNeedleRegExp($needle);
+
+        static::assertSame(
+            '/' . preg_quote($needle, '/') . '/',
+            str_replace('\\s*+', '', $result)
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function getCssNeedleRegExpNotEscapesNonSpecialCharacters()
+    {
+        $needle
+            = implode('', range('a', 'z'))
+            . implode('', range('A', 'Z'))
+            . implode('', range('0 ', '9 '))
+            . "\r\n\t `¬\"£%&_;'@#~,";
+
+        $result = TestingAssertCss::getCssNeedleRegExp($needle);
+
+        static::assertSame(
+            '/' . $needle . '/',
+            str_replace('\\s*+', '', $result)
+        );
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function contentWithOptionalWhitespaceDataProvider()
+    {
+        return [
+            '"{" alone' => ['{', ''],
+            '"}" alone' => ['}', ''],
+            '"{" with non-special character' => ['{', 'a'],
+            '"{" with two non-special characters' => ['{', 'a0'],
+            '"{" with special character' => ['{', '.'],
+            '"{" with two special characters' => ['{', '.+'],
+            '"{" with special character and non-special character' => ['{', '.a'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string $contentToInsertAround
+     * @param string $otherContent
+     *
+     * @dataProvider contentWithOptionalWhitespaceDataProvider
+     */
+    public function getCssNeedleRegExpInsertsOptionalWhitespace($contentToInsertAround, $otherContent)
+    {
+        $result = TestingAssertCss::getCssNeedleRegExp(
+            $otherContent . $contentToInsertAround . $otherContent
+        );
+
+        $quotedOtherContent = preg_quote($otherContent, '/');
+
+        static::assertSame(
+            '/' . $quotedOtherContent . '\\s*+' . preg_quote($contentToInsertAround, '/') . '\\s*+'
+                . $quotedOtherContent . '/',
```

Comment:
Could you please extract this to a variable `$expectedContent`?

Reply:
Ok